### PR TITLE
feat: set "PYCSW_SERVER_URL" env

### PR DESCRIPTION
**Pull request by heliprokupetz14**
This PR sets the environment "PYCSW_SERVER_URL" to be the actual route instead of "localhost:8000".
Use Cases:
1. If the host is specified, the env will be the route (the actual route that was deployed) according to these values: "https", "host" and "path".
2. If the host is not specified, the env will remain as localhost.

### CODE REVIEW

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 1]
```diff
@@ -59,7 +59,6 @@ Selector labels
 app.kubernetes.io/name: {{ include "pycsw.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
-
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:62]: Remove this unnecessary change

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 2]
```diff
@@ -70,7 +69,28 @@ Returns the environment from global if exists or from the chart's values, defaul
     {{- .Values.environment | default "development" -}}
 {{- end -}}
 {{- end -}}
-
+{{/*
+  Set the PYCSW_SERVER_URL environment variable based on the route of pycsw.
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:73]: Fix this description. It is written as a mission. You can see other functions in this file to see some good examples

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 3]
```diff
@@ -70,7 +69,28 @@ Returns the environment from global if exists or from the chart's values, defaul
...
+  If no host is provided, set the environment to "localhost".
+  If HTTPS is false, prepend "http://".
+  If HTTPS is true, prepend "https://".
+  If path is provided, append it to the URL.
+*/}}
+{{- define "pycsw-setServerURL" -}}
+  {{- $serverURL := "" }}
+  {{- with .Values.route -}}
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:81]: Remove this "with" option. It makes the code less readable. It is better to write ".Values.route.VALUE"

File: charts/serving/charts/pycsw/templates/deployment.yaml [hunk 4]
```diff
@@ -5,6 +5,7 @@
 {{- $cloudProviderImagePullSecretName := include "pycsw.cloudProviderImagePullSecretName" . -}}
 {{ $caSecretName := ternary .Values.global.ca.secretName (printf "%s%s" .Release.Name "-root-ca") .Values.global.ca.useExternal }}
 {{- $imageTag := include "pycsw.tag" . -}}
+{{- $route := include "pycsw-setServerURL" . -}}
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/deployment.yaml:8]: replace "route" var with a better name. Maybe "serverURL".

Same for "pycsw-setServerURL", try to keep the exist design ("pycsw." and not "pycsw-")

File: values-dev.yaml [hunk 5]
```diff
@@ -9,7 +9,7 @@ global:
         jobSyncer: true
         catalog: true
     serving:
-      enabled: false
+      enabled: true
```
Comment (netanelC) [inline values-dev.yaml:12]: Why did you change it?

File: values-dev.yaml [hunk 6]
```diff
@@ -581,9 +581,9 @@ serving:
 
     route:
       enabled: true
-      host:
-      path: /api/3d/v1
-      https: true
+      host: 10.0.0.4
```
Comment (netanelC) [inline values-dev.yaml:584]: Why did you change this line and the next lines? values should be with no changes

**Review by netanelC — CHANGES_REQUESTED** [review]

File: charts/serving/charts/pycsw/templates/_helpers.tpl (hunk 3 above)
Comment (heliprokupetz14) [reply charts/serving/charts/pycsw/templates/_helpers.tpl:81]: the with is to use the values from the route...

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 7]
```diff
@@ -71,6 +71,24 @@ Returns the environment from global if exists or from the chart's values, defaul
...
+    {{- else -}}
+      {{- $protocol := .Values.route.https | ternary "https" "http" }}
+      {{- $serverURL = printf "%s://%s" $protocol .Values.route.host }}
+      {{- if .Values.route.path }}
+        {{- $serverURL = printf "%s%s" $serverURL .Values.route.path }}
+      {{- end -}}
+    {{- end -}}
+  {{- end -}}
+  {{- printf "%s" $serverURL | quote }}
+{{- end -}}
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:91]: Add an empty line to separate the functions

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 8]
```diff
@@ -71,6 +71,24 @@ Returns the environment from global if exists or from the chart's values, defaul
 {{- end -}}
 {{- end -}}
 
+{{/*
+  Returns the pycsw server url based on the provided values.  
+*/}}
+{{- define "pycsw.setServerURL" -}}
+  {{- $serverURL := "" }}
+  {{- if .Values.route -}}
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:79]: Always true. Why did you write this?

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 9]
```diff
@@ -71,6 +71,24 @@ Returns the environment from global if exists or from the chart's values, defaul
 {{- end -}}
 {{- end -}}
 
+{{/*
+  Returns the pycsw server url based on the provided values.  
+*/}}
+{{- define "pycsw.setServerURL" -}}
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:77]: Use better name:
```suggestion
{{- define "pycsw.serverURL" -}}
```

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 10]
```diff
@@ -71,6 +71,24 @@ Returns the environment from global if exists or from the chart's values, defaul
...
+  {{- $serverURL := "" }}
+  {{- if .Values.route -}}
+    {{- if not .Values.route.host }}
+      {{- $serverURL = printf "http://localhost:8000" }}
+    {{- else -}}
+      {{- $protocol := .Values.route.https | ternary "https" "http" }}
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:83]: That is not a good practice for using ternary.

File: charts/serving/charts/pycsw/templates/_helpers.tpl [hunk 11]
```diff
@@ -71,6 +71,24 @@ Returns the environment from global if exists or from the chart's values, defaul
...
+      {{- $serverURL = printf "%s://%s" $protocol .Values.route.host }}
+      {{- if .Values.route.path }}
```
Comment (netanelC) [inline charts/serving/charts/pycsw/templates/_helpers.tpl:85]: Is this necessary? 
If path is empty, I don't care about putting him as well

**Review by netanelC — CHANGES_REQUESTED** [review]

File: charts/serving/charts/pycsw/templates/_helpers.tpl (hunk 11 above)
Comment (heliprokupetz14) [reply charts/serving/charts/pycsw/templates/_helpers.tpl:85]: yes it is necessary.

**Review by netanelC — CHANGES_REQUESTED** [review]
Remove values changes

File: charts/serving/charts/pycsw/templates/_helpers.tpl (hunk 7 above)
Comment (shimoncohen) [reply charts/serving/charts/pycsw/templates/_helpers.tpl:89]: @heliprokupetz14 Do not resolve anything opened on something you did, you let the person who left the comment review and decide if you finished.

**Review by shimoncohen — APPROVED** [review]

File: charts/serving/charts/pycsw/templates/_helpers.tpl (hunk 7 above)
Comment (netanelC) [reply charts/serving/charts/pycsw/templates/_helpers.tpl:89]: @heliprokupetz14 You didn't add an empty line.

Comment (shimoncohen) [reply charts/serving/charts/pycsw/templates/_helpers.tpl:89]: This proves my point.

**Review by netanelC — APPROVED** [review]

**Pull request merged**